### ISSUE TITLE
Accessibility-> Users using Screen reader won't know if no results found from their search until then navigate through the full page

### ISSUE THREAD

**Issue by manuji** [bcgov/nr-rec-resources#660]
**Describe the bug**
Users using the screen reader won't get notified about the no results for their search when searching for a site or trail. The only way that user would know is by navigating through the whole page until they get to the No results text. This issue fails the [4.1.3 Status Message (Level AA)](https://www.w3.org/WAI/WCAG22/Understanding/status-messages) criterion. 

**To Reproduce**
1. Go to 'https://d1vrz2gt4nmfe5.cloudfront.net/'
2. Then navigate to the Find a site or trail page
3. While using the screen reader search for a keyword that do not have any results. 
4. Check if the user get informed about the no results.

**Expected behavior**
The user should be informed about the no results without having to navigate through the full page. 

**Screenshots**

https://github.com/user-attachments/assets/5805a9ed-d7ff-469b-b5e8-7f685dbc66ae

**Additional context**
Tested on MacOS with Chrome and VoiceOver.

**Comment by airliah** [comment]
@marcellmueller for me after I type in a search that gives me no results, and I hit 'enter', it highlights the link at the bottom of the text ('go back to full list'). but it should go to the top of the text saying '0 Results containing 'asdf' so that users know there are no results and can then tab through to the 'go back' link. is that doable?

**Comment by marcellmueller** [comment]
> [@marcellmueller](https://github.com/marcellmueller) for me after I type in a search that gives me no results, and I hit 'enter', it highlights the link at the bottom of the text ('go back to full list'). but it should go to the top of the text saying '0 Results containing 'asdf' so that users know there are no results and can then tab through to the 'go back' link. is that doable?

You're only supposed to focus on interactive elements, not text...bcparks.ca focuses on that button as well.

**Comment by airliah** [comment]
@manuji when you have a moment can you review this and let us know if it's working as expected? We will need to launch this tomorrow but please write up another bug if we need to do more work on this! Thanks

**Comment by manuji** [comment]
@airliah , I would like to re-open this issue as it hasn't been fixed. When using the screenreader, the user doesn't get notified about the no results when searching. Please refer to BCParks.ca for example while having the screenreader on.

**Comment by airliah** [comment]
@marcellmueller please see Manuji's comment and provide the necessary fixes/improvements. thanks!
